### PR TITLE
add Laravel 10 support

### PR DESCRIPTION
**Pull request by jonnott**
## Description of the change

Add Laravel 10 support

## Type of change

Maintenance

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     ],
     "require": {
         "php": ">=7.2|^8.0",
```
Comment (fh32000) [inline composer.json:25]: I think You should  add php 8.1 
 ```
 "php": ">=7.2|^8.02|^8.1",
 ```

Comment (jonnott) [reply composer.json:25]: Yes, agreed, merging this will also need to allow PHP 8.1 so it can be used with Laravel 10, which will require PHP 8.1.

**Comment by jonnott** [comment]
@cyrusradfar @danielmorell  .. or anyone currently at Rollbar .. can you hear us?

File: composer.json (hunk 1 above)
Comment (danielmorell) [reply composer.json:25]: There is no nee do update the PHP version constraint. PHP 8.1 is already allowed. For reference see the [composer docs](https://getcomposer.org/doc/articles/versions.md#caret-version-range-).

**Review by danielmorell — APPROVED** [review]
Looks good! Thank you for your help @jonnott!!

**Comment by jonnott** [comment]
Thanks @danielmorell .. how to we move this to getting a new release tagged, seeing as Laravel 10.0.0 is now released?

**Comment by michaelscola** [comment]
I'm not sure how this would be fully working, as Laravel 10 upgrades to Monolog 3 and the current installed version only does Monolog 2. I will be doing a pull request later for both rollbar-php-laravel and rollbar-bar php to address this.

**Comment by jonnott** [comment]
> I'm not sure how this would be fully working, as Laravel 10 upgrades to Monolog 3 and the current installed version only does Monolog 2. I will be doing a pull request later for both rollbar-php-laravel and rollbar-bar php to address this.

@michaelscola Do you mean a bit like this PR? https://github.com/rollbar/rollbar-php/pull/600

**Comment by michaelscola** [comment]
> > I'm not sure how this would be fully working, as Laravel 10 upgrades to Monolog 3 and the current installed version only does Monolog 2. I will be doing a pull request later for both rollbar-php-laravel and rollbar-bar php to address this.
> 
> @michaelscola Do you mean a bit like this PR? [rollbar/rollbar-php#600](https://github.com/rollbar/rollbar-php/pull/600)

@jonnott I do mean like that PR, except the Monolog handler in the rollbar-php-laravel is not going to work properly when Monolog 3 is installed. The write method has an updated typehint for LogRecord instead of array.

**Comment by jonnott** [comment]
> @jonnott I do mean like that PR, except the Monolog handler in the rollbar-php-laravel package is not going to work properly when Monolog 3 is installed. The write method has an updated typehint for LogRecord instead of array.

Please do add further commits/PR to resolve that problem.

**Comment by jonnott** [comment]
@danielmorell I think this PR will need to be re-opened/added and we'll need a new major version of this package (BC break) and possibly also a new major version of the underlying rollbar-php package too. 

Laravel 10 requires monolog v3, which rollbar-php doesn't support (it only supports v2), and also as @michaelscola points out, there's a type change to the argument on the monolog/RollbarHandler.php write() method requiring a Monolog\LogRecord instance.

**Comment by michaelscola** [comment]
@jonott this absolutely is a breaking change, I'm currently testing the change myself in development and I will add a new pull request once I can confirm everything is working correctly.

**Comment by danielmorell** [comment]
Thank you for pointing out the monolog v3 conflict!

I am just finishing up upgrading https://github.com/rollbar/rollbar-php to support monolog v3. The last part has been updating the test suite. I should have a PR out tomorrow. I will tag this thread once I do.

**Comment by jonnott** [comment]
> Thank you for pointing out the monolog v3 conflict!

No worries.

I guess there'll need to be a new (major, breaking) version of the base package tagged before this package can have the dependency raised to that new version before a new major (also breaking) release can be tagged here.

**Comment by michaelscola** [comment]
@danielmorell @jonnott I have a new pull request for the updates for Monolog. Some of the properties that you overwrite are Readonly and ->with has unexpected behavior, so it was easier to create a new Instance with the additional data rollbar wants to add. I have tested it and everything appears to be working as expected, look forward to hearing from you with your comments.

**Comment by jonnott** [comment]
@danielmorell Do you know for what reason 8.0.0 isn't showing up on packagist.org yet? It's been tagged for 8 hours now.

**Comment by jonnott** [comment]
> Hmm. Let me see if I can fix that.

You fixed it! 👍

**Pull request merged**